torchtune/data/_prompt_templates.py

```diff
@@ -110,11 +110,22 @@ def __call__(
             if message.role in self.template:
                 prepend_tag = self.template[message.role][0]
                 append_tag = self.template[message.role][1]
-                content = (
-                    [{"type": "text", "content": prepend_tag}]
-                    + message.content
-                    + [{"type": "text", "content": append_tag}]
-                )
+                if isinstance(prepend_tag, str) and isinstance(append_tag, str):
```

Comment:
maybe we should structure this slightly different to avoid the cascade of ifs:
```
content = message.content
if message.role in self.template:
    prepend_tag = self.template[message.role][0]
    append_tag = self.template[message.role][1]
    if prepend_tag is not None and len(prepend_tag) > 0:
        # add prepend tag
    if append_tag is not None and len(append_tag) > 0:
        # add append tag
formatted_dialogue.append(Message(...))
```

Reply:
It depends on if can something "bad" except None can come there

Reply:
Ah, I see now how do you want to do it. Ok, let me fix it like this. But I would like to save isinstance there